sources/web/datalab/settings.ts

```diff
@@ -396,6 +399,21 @@ export function createHandler(): http.RequestHandler {
   return requestHandler;
 }
 
+function renameBadUserSettings(settingsPath: string) {
+  let newPath = settingsPath + ".bad";
+  let n = 0;
+  while (fs.existsSync(newPath)) {
```

Comment:
I'm not too worried about filling up the disk, these are small files. I'm more concerned about losing the user's settings file if we always move to the same file. I put in a limit of ten backups.